### PR TITLE
Fix various bugs

### PR DESCRIPTION
**Pull request by e5l**

### CODE REVIEW

**Comment by e5l** [comment]
Sure

File: core/commonMain/src/kotlinx/io/text/TextInputObsolete.kt [hunk 1]
```diff
@@ -127,7 +131,7 @@ private inline fun Input.decodeUtf8Chars(consumer: (Char) -> Boolean): Int {
                             byte < 0x80 -> {
                                 if (!consumer(byte.toChar())) {
                                     state = STATE_FINISH
-                                    return@readBufferRange offset + 1
+                                    return@readBufferRange offset - startOffset + 1
```
Comment (qwwdfsad) [inline core/commonMain/src/kotlinx/io/text/TextInputObsolete.kt]: This line is not covered by any test

File: core/commonMain/src/kotlinx/io/text/TextInputObsolete.kt [hunk 2]
```diff
@@ -188,15 +192,15 @@ private inline fun Input.decodeUtf8Chars(consumer: (Char) -> Boolean): Int {
                             }
                             if (!more) {
                                 state = STATE_FINISH
-                                return@readBufferRange offset + 1
+                                return@readBufferRange offset - startOffset + 1
```
Comment (qwwdfsad) [inline core/commonMain/src/kotlinx/io/text/TextInputObsolete.kt]: this either

File: core/commonMain/src/kotlinx/io/Input.kt [hunk 3]
```diff
@@ -136,9 +136,10 @@ public abstract class Input : Closeable {
 
             return readBufferRange { buffer, startOffset, endOffset ->
                 destination.writeBuffer(buffer, startOffset, endOffset)
-                endOffset
+                endOffset - startOffset
```
Comment (qwwdfsad) [inline core/commonMain/src/kotlinx/io/Input.kt]: And this one is not tested

**Pull request merged**